src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -1699,6 +1699,31 @@ public IReadOnlyList<SyntaxNode> GetStatementContainerStatements(SyntaxNode node
                ? GetExecutableBlockStatements(node)
                : (IReadOnlyList<SyntaxNode>)ImmutableArray.Create<SyntaxNode>(node.GetEmbeddedStatement());
 
+        public bool IsTerminator(SyntaxNode node, SyntaxToken token)
+        {
+            if (token.IsKind(SyntaxKind.SemicolonToken))
+            {
+                if (node.IsKind(SyntaxKind.UsingDirective, out UsingDirectiveSyntax usingDirective))
+                {
+                    return usingDirective.SemicolonToken == token;
+                }
+                else if (node.IsKind(SyntaxKind.ExternAliasDirective, out ExternAliasDirectiveSyntax externAliasDirective))
+                {
+                    return externAliasDirective.SemicolonToken == token;
+                }
+                else
+                {
+                    // TODO: handle other terminating semicolons
+                    return false;
+                }
+            }
+            else
+            {
+                // TODO: handle other terminating tokens
+                return false;
+            }
```

Comment:
not a huge fan of this method, primarily because i don't have an intuition about what it means to be a 'terminator'.  besides: 

![image](https://user-images.githubusercontent.com/4564579/82613456-2bb7c280-9b7a-11ea-9e28-25dc49c6b30a.png)

Reply:
➡️ Covered by https://github.com/dotnet/roslyn/pull/44402#discussion_r428955803